### PR TITLE
feat(web): Adding datadog-rum to islandis web

### PR DESCRIPTION
**Pull request by dabbeg**
# ...

Closes: https://github.com/island-is/infrastructure/issues/839

Try interacting with the feature deployment here [https://datadog-rum-beta.dev01.devland.is/](https://datadog-rum-beta.dev01.devland.is/)

And then take a look at the dashboard [https://app.datadoghq.eu/rum/application/665a387d-16c4-4c86-88c8-e059d8120021/overview/browser?from_ts=1639489649100&live=true&query=env%3Adev+service%3Aislandis&to_ts=1639576049100](https://app.datadoghq.eu/rum/application/665a387d-16c4-4c86-88c8-e059d8120021/overview/browser?from_ts=1639489649100&live=true&query=env%3Adev+service%3Aislandis&to_ts=1639576049100)

## What

Try out the Datadog real user montoring.

This PR enables datadog rum for islandis web, if we find this useful we can enable this for more projects.

## Why

Trying out datadog rum service

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/user-monitoring/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# User Monitoring
+
+This library is to encapsulate
```
Comment (pshomov) [inline libs/user-monitoring/README.md:3]: Encapsulate .....

**Review by pshomov — COMMENTED** [review]
Looking good!

File: libs/infra-tracing/src/lib/datadog-tracer.spec.ts [hunk 2]
```diff
@@ -33,33 +33,4 @@ describe('datadogTrace', () => {
...
-      process.env.DD_PII_MASKING_DISABLED_ON_FAILURE = 'true'
-      const ddtrace = require('./datadog-tracer') //eslint-disable-line
-
-      ;[
-        'http://island.is/path/to/0000000000',
-        'http://island.is/path/to/1234567890/',
-        'http://island.is/path/to/5864239878/boiah',
-      ].forEach((url) => expect(ddtrace.rewriteUrl(url)).toEqual(url))
-    })
-  })
```
Comment (pshomov) [inline libs/infra-tracing/src/lib/datadog-tracer.spec.ts:64]: Why remove this?

**Review by pshomov — COMMENTED** [review]
Most of the changes are good but it seems like there is a part of a different PR in there?

File: libs/infra-tracing/src/lib/datadog-tracer.spec.ts (hunk 2 above)
Comment (dabbeg) [reply libs/infra-tracing/src/lib/datadog-tracer.spec.ts:64]: There are currently two different functions that mask nationalIds in the codebase (logging library and datadog-tracer library) and I needed to create the third one.

So instead I created the library `libs/shared/pii` which has one function `maskNationalId` which I use in all three places.

File: libs/shared/pii/README.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# Shared Pii
+
+This library was generated with [Nx](https://nx.dev).
```
Comment (eirikurn) [inline libs/shared/pii/README.md:3]: Maybe a short description for gitbook?

File: libs/user-monitoring/src/lib/user-monitoring.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+interface DdRumInitParams {
+  env: string
+  version: string
+  service: string
+}
+
+const initDdRum = (params: DdRumInitParams) => {
+  datadogRum.init({
+    applicationId: '665a387d-16c4-4c86-88c8-e059d8120021',
+    clientToken: 'pub7a898a272110329d3a2e73e09e263f5b',
```
Comment (eirikurn) [inline libs/user-monitoring/src/lib/user-monitoring.ts:14]: Do we want to hard code this here?

File: libs/user-monitoring/src/lib/user-monitoring.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    service: params.service,
+    env: params.env,
+    version: params.version,
+    sampleRate: 100,
+    trackInteractions: true,
+    allowedTracingOrigins: [
+      'https://island.is',
+      /https:\/\/.*\.island\.is/,
+      /https:\/\/.*\.devland\.is/,
+      'http://localhost:4200',
```
Comment (eirikurn) [inline libs/user-monitoring/src/lib/user-monitoring.ts:25]: Maybe more localhost ports for future apps?

**Review by eirikurn — COMMENTED** [review]
I don't like collecting more and more third party scripts for performance reasons.

Happy to approve this one if it can replace Sentry. :)

**Comment by eirikurn** [comment]
Also, nice `maskNationalId` refactor!

File: libs/user-monitoring/src/lib/user-monitoring.ts (hunk 4 above)
Comment (dabbeg) [reply libs/user-monitoring/src/lib/user-monitoring.ts]: No probably not, I moved this to environment variables.

**Review by pshomov — APPROVED** [review]
LGTM

**Review by stjanilofts — APPROVED** [review]

**Review by eirikurn — APPROVED** [review]

**Pull request merged**